snapcraft/tests/test_plugin_catkin.py

```diff
@@ -183,3 +331,136 @@ class Options:
+                         'Error initializing rosdep database: bar')
+
+    def test_setup_update_failure(self):
+        def run(args, **kwargs):
+            if args == ['rosdep', 'update']:
+                raise subprocess.CalledProcessError(1, 'foo', 'bar')
+
+            return mock.DEFAULT
+
+        self.run_mock.side_effect = run
+
+        with self.assertRaises(RuntimeError) as raised:
+            self.rosdep.setup()
+
+        self.assertEqual(raised.exception.args[0],
+                         'Error updating rosdep database: bar')
+
+    def test_get_dependencies(self):
+        self.run_mock.return_value = b'foo\nbar\nbaz'
+
+        self.assertEqual(self.rosdep.get_dependencies('foo'),
+                         ['foo', 'bar', 'baz'])
+
+        self.run_mock.assert_called_with(['rosdep', 'keys', 'foo'],
+                                         env=mock.ANY)
+
+    def test_get_dependencies_no_dependencies(self):
+        self.run_mock.return_value = b''
+
+        self.assertEqual(self.rosdep.get_dependencies('foo'), [])
+
+    def test_get_dependencies_invalid_package(self):
+        self.run_mock.side_effect = subprocess.CalledProcessError(1, 'foo')
+
+        with self.assertRaises(FileNotFoundError) as raised:
+            self.rosdep.get_dependencies('bar')
+
+        self.assertEqual(raised.exception.args[0],
+                         'Unable to find Catkin package "bar"')
+
+    def test_resolve_dependency(self):
+        self.run_mock.return_value = b'#apt\nmylib-dev'
+
+        self.assertEqual(self.rosdep.resolve_dependency('foo'), 'mylib-dev')
+
+        self.run_mock.assert_called_with(['rosdep', 'resolve', 'foo',
+                                          '--rosdistro', 'ros_distro'],
+                                         env=mock.ANY)
+
+    def test_resolve_invalid_dependency(self):
+        self.run_mock.side_effect = subprocess.CalledProcessError(1, 'foo')
+
+        self.assertEqual(self.rosdep.resolve_dependency('bar'), None)
+
+    def test_resolve_dependency_weird_output(self):
+        self.run_mock.return_value = b'mylib-dev'
+
+        with self.assertRaises(RuntimeError) as raised:
+            self.rosdep.resolve_dependency('')
+
+        self.assertEqual(raised.exception.args[0],
+                         'Unexpected rosdep resolve output:\nmylib-dev')
+
+    def test_run(self):
+        rosdep = self.rosdep
+        rosdep._run(['qux'])
+
+        class check_env():
+            def __eq__(self, env):
+                return (
+                    env['PATH'] == os.path.join(rosdep._rosdep_install_path,
+                                                'usr', 'bin') and
+                    env['PYTHONPATH'] == os.path.join(
+                        rosdep._rosdep_install_path, 'usr', 'lib', 'python2.7',
+                        'dist-packages') and
+                    env['ROSDEP_SOURCE_PATH'] == rosdep._rosdep_sources_path
+                    and env['ROS_HOME'] == rosdep._rosdep_cache_path and
```

Comment:
the and operator should be at the end of the line. flake8 complaints in xenial, it seems to ignore it in the travis's version.